### DISCUSSION TITLE
Unable to filter interface-templates by device type (slug or id)

### DISCUSSION THREAD

**Discussion by nonPointer** [netbox-community/netbox#14509] [Q&A] [answered]
Endpoint: `api/dcim/interface-templates`
Neither device type ID nor device type slug not working.
I encountered the same issue as [this post](https://www.reddit.com/r/Netbox/comments/13uob63/unable_to_filter_api_results/) on 
NetBox v3.6.6. It seems to be an implementation issue of the API endpoint. Thanks!

**Comment by candlerb** [comment, marked as answer]
For completeness, can you show the **exact** query parameters you're sending?

Looking at the source code, it seems that `devicetype_id` (with no underscore between `device` and `type`) is the correct name of the filter.

```
class InterfaceTemplateFilterSet(ChangeLoggedModelFilterSet, ModularDeviceTypeComponentFilterSet):
    type = django_filters.MultipleChoiceFilter(
        choices=InterfaceTypeChoices,
        null_value=None
    )
    bridge_id = django_filters.ModelMultipleChoiceFilter(
        field_name='bridge',
        queryset=InterfaceTemplate.objects.all()
    )
    poe_mode = django_filters.MultipleChoiceFilter(
        choices=InterfacePoEModeChoices
    )
    poe_type = django_filters.MultipleChoiceFilter(
        choices=InterfacePoETypeChoices
    )
    rf_role = django_filters.MultipleChoiceFilter(
        choices=WirelessRoleChoices
    )

///

class ModularDeviceTypeComponentFilterSet(DeviceTypeComponentFilterSet):
    moduletype_id = django_filters.ModelMultipleChoiceFilter(
        queryset=ModuleType.objects.all(),
        field_name='module_type_id',
        label=_('Module type (ID)'),
    )

///

class DeviceTypeComponentFilterSet(django_filters.FilterSet):
    q = django_filters.CharFilter(
        method='search',
        label=_('Search'),
    )
    devicetype_id = django_filters.ModelMultipleChoiceFilter(    // <<<<<<<< NOTE
        queryset=DeviceType.objects.all(),
        field_name='device_type_id',
        label=_('Device type (ID)'),
    )
```

Also you can see this in the REST API documentation, which you can get from a link on the bottom of the Netbox front screen:

![image](https://github.com/netbox-community/netbox/assets/44789/c40d41a6-ba30-4124-9403-ac2fc3795ad1)

**Reply by nonPointer** [reply]
Thanks [candlerb](https://github.com/candlerb) for your response.
I tried `device-type-id`, which should be `devicetype_id`. Now it works! The built-in Swagger is very helpful.
But for endpoint `api/dcim/devices`, it uses `device-type-id`. The parameter naming schema of NetBox API seems not consistent.
Many thanks!

**Reply by candlerb** [reply]
> But for endpoint `api/dcim/devices`, it uses `device-type-id`

Actually I believe it uses `device_type_id`. [Source](https://github.com/netbox-community/netbox/blob/v3.6.6/netbox/dcim/filtersets.py#L845-L848)

> The parameter naming schema of NetBox API seems not consistent.

An oversight no doubt. If it were changed now, it would affect users with existing integrations. (Netbox *does* make backwards-incompatible API changes fairly regularly though, and if you'd like to see this changed, you can raise a feature request)

FWIW, the content type name is `dcim.devicetype` and the postgres table is `dcim_devicetype`

**Reply by nonPointer** [reply]
Thanks for the clarification!
